src/main/java/org/broadinstitute/consent/http/service/SupportRequestService.java

```diff
@@ -0,0 +1,91 @@
+
+import javax.ws.rs.ServerErrorException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class SupportRequestService {
+
+    private final HttpClientUtil clientUtil;
+    private final ServicesConfiguration configuration;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Inject
+    public SupportRequestService(ServicesConfiguration configuration) {
+        this.clientUtil = new HttpClientUtil();
+        this.configuration = configuration;
+    }
+
+    /**
+     * Builds a ticket with the proper structure to request support via Zendesk
+     * @param name The name of the user requesting support
+     * @param type The type of request ("question", "incident", "problem", "task")
+     * @param email The email of the user requesting support
+     * @param subject Subject line of the request
+     * @param description Description of the task or question
+     * @param url The API url of this request
+     * @return A structured ticket used to make the request
+     */
+    public SupportTicket createSupportTicket(String name, String type, String email, String subject, String description, String url) {
+        if (Objects.isNull(name) || Objects.isNull(email)) {
+            throw new IllegalArgumentException("Name and email of user requesting support is required");
+        }
+        if (Objects.isNull(subject)) {
+            throw new IllegalArgumentException("Support ticket subject is required");
+        }
+        if (Objects.isNull(description)) {
+            throw new IllegalArgumentException("Support ticket description is required");
+        }
+        if (Objects.isNull(type)) {
+            throw new IllegalArgumentException("Support ticket type is required");
+        }
+        if (Objects.isNull(url)) {
+            throw new IllegalArgumentException("Support ticket url is required");
+        }
+
+        SupportRequester requester = new SupportRequester(name, email);
+        List<CustomRequestField> customFields = new ArrayList<>();
+        customFields.add(new CustomRequestField("360012744452", type));
+        customFields.add(new CustomRequestField("360007369412", description));
+        customFields.add(new CustomRequestField("360012744292", name));
+        customFields.add(new CustomRequestField("360012782111", email));
+        customFields.add(new CustomRequestField("360018545031", email));
+        SupportRequestComment comment = new SupportRequestComment(description, url);
+
+        return new SupportTicket(requester, subject, customFields, comment);
+    }
+
+    public void postTicketToSupport(SupportTicket ticket, AuthUser authUser) throws Exception {
+        GenericUrl genericUrl = new GenericUrl(configuration.postSupportRequestUrl());
+        JsonHttpContent content = new JsonHttpContent(new GsonFactory(), ticket);
```

Comment:
I think this is where we need to do some extra work. Standard Gson conversion of the object will camel-case the field names. I was able to get the right format with this code:
```
System.out.println(new GsonBuilder()
    .setPrettyPrinting()
    .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
    .create()
    .toJson(ticket));
```
but then we need to somehow use the naming policy in `GsonFactory` which I don't think we can do. Looking into workarounds in the meantime.

Reply:
If I use this in the test: 
``` 
 mockServerClient.when(request())
                .respond(response()
                        .withHeader(Header.header("Content-Type", "application/json"))
                        .withStatusCode(HttpStatusCodes.STATUS_CODE_OK));

 service.postTicketToSupport(ticket, authUser);
 LogEventRequestAndResponse[] requests = mockServerClient.retrieveRecordedRequestsAndResponses(null);
```
to see all requests and responses but not require a specific request body for `mockServerClient `to work, then this is the request sent:
``` {
  "method" : "POST",
  "path" : "/",
  "headers" : {
    "x-app-id" : [ "DUOS" ],
    "content-length" : [ "435" ],
    "User-Agent" : [ "Google-HTTP-Java-Client/1.42.0 (gzip)" ],
    "Host" : [ "localhost:53764" ],
    "Content-Type" : [ "application/json; charset=UTF-8" ],
    "Connection" : [ "keep-alive" ],
    "Authorization" : [ "Bearer null" ],
    "Accept-Encoding" : [ "gzip" ],
    "Accept" : [ "application/json" ]
  },
  "keepAlive" : true,
  "secure" : false,
  "remoteAddress" : "172.17.0.1",
  "body" : "{\n  \"request\": {\n    \"requester\": {\n      \"name\": \"nVuVnlRfWH\",\n      \"email\": \"ppgJuTWFkI\"\n    },\n    \"subject\": \"PDctpZQwpK\",\n    \"custom_fields\": [\n      {\n        \"id\": 6428364382081724521,\n        \"value\": \"JXwYyyoKmM\"\n      }\n    ],\n    \"comment\": {\n      \"body\": \"WahAaCBerC\\n\\n------------------\\nSubmitted from: FVYhtibdrv\"\n    },\n    \"ticket_form_id\": 360000669472\n  }\n}"
}
```